python/urlcanon/rules.py

```diff
@@ -67,15 +86,41 @@ def url_matches_domain(url, domain):
         url = urlcanon.parse_url(url)
     return host_matches_domain(url.host, domain)
 
+
+def url_matches_domain_exactly(url, domain):
+    '''
+    Returns true if
+     - domain is an ip address and url.host is the same ip address
+     - domain is a domain and url.host is the same domain
+
+    Does not do any normalization/canonicalization. Probably a good idea to
+    call `host_matches_domain(
+            canonicalize(url), urlcanon.normalize_host(domain))`.
+    '''
+    if not isinstance(url, urlcanon.ParsedUrl):
+        url = urlcanon.parse_url(url)
+    return host_matches_domain_exactly(url.host, domain)
+
+
 class MatchRule:
     '''
     A url-matching rule, with one or more conditions.
 
     All conditions must match for a url to be considered a match.
 
-    The supported conditions are `surt`, `ssurt`, `regex`, `domain`,
-    `substring`, `parent_url_regex`. Values should be bytes objects. If they
-    are unicode strings, they will be utf-8 encoded.
+    The supported conditions are `surt`, `ssurt`, `regex`, `domain`
+    `substring`, `parent_url_regex`. Values should be bytes objects.
+    If they are unicode strings, they will be utf-8 encoded.
+
+    The supported condition modifiers `exact`.
+
+    The `exact` modifier switches the rule into matching the supplied url
+    against the configured conditions exactly, that is to say the domain,
+    surt, and or ssurts of a supplied url must match exactly to the values
+    the rule was configured with.
+
+    Note: The `regex` or `substring` conditions are not affected by
+    either of the condition modifiers.
```

Comment:
Remove "either of" since there's only one at the moment?

Reply:
Done. 
This wording was a remnant of an previous iteration where there were additional modifiers.
Thanks for pointing this out.